Fixed: "Argument 1 passed to Friendica\Util\Temporal::getRelativeDate() must be of the type string, null given"

Quix0r opened the pull request:
This is a rebased change on current `develop` code. The original ticket was #11900 .

Quix0r committed:
Changed:
- first parameter can be null

Quix0r committed:
Changed according feedback:
- $posted_date should not allow NULL, instead use `?? ''` instead
- this happened in e.g. 2FA app-specific password listing

MrPetovan commented:
Please rebase on the latest `develop` to take into account `messages.po` and `database.sql` changes.

MrPetovan commented:
I thought we agreed on keeping the null value possibility and testing for it first? https://github.com/friendica/friendica/pull/11900#discussion_r979248182

This would remove the need for the null coalescing operator.

Quix0r committed:
Changed:
- re-added `null` again https://github.com/friendica/friendica/pull/11900#discussion_r979248182
- removed null coalescing operator

Quix0r commented:
I was confused there. :roll_eyes:
